File: app/src/main/java/com/openclassrooms/realestatemanager/ui/createAndEditEstate/PhotoAdapter.java
package com.openclassrooms.realestatemanager.ui.createAndEditEstate;

import android.net.Uri;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.RequestManager;
import com.openclassrooms.realestatemanager.databinding.ActivityAddPhotoItemBinding;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class PhotoAdapter extends RecyclerView.Adapter<PhotoViewHolder> {
    private RequestManager glide;
    private List<Uri> mPhotoList = new ArrayList<Uri>();
    private List<String> mPhotoDescription;
    private boolean detail;

    /**
     * Constructor
     *
     * @param glide
     * @param photoDescription
     * @param detail
     */
    public PhotoAdapter( RequestManager glide, ArrayList<String> photoDescription, boolean detail) {
        this.glide = glide;
        this.mPhotoDescription = photoDescription;
        this.detail = detail;
    }

    @NonNull
    @Override
    public PhotoViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        return new PhotoViewHolder(ActivityAddPhotoItemBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    /**
     * Update viewHolder
     *
     * @param holder
     * @param position
     */
    @Override
    public void onBindViewHolder(@NonNull PhotoViewHolder holder, int position) {
        //For photo description with photo
        Uri photoUri = null;
        String photoDescription = "";

        if (mPhotoList.size() > position) {
            photoUri = mPhotoList.get(position);
        }

        if (mPhotoDescription.size() > position) {
            photoDescription = mPhotoDescription.get(position);
        }
        try {
            holder.updateWithDetails(photoUri, this.glide, photoDescription,detail);
        } catch (Exception e) {
            e.getMessage();
        }
    }

    /**
     * For return photos
     *
     * @return
     */
    @Override
    public int getItemCount() {
        return mPhotoList.size();
    }

    /**
     * for set photolist & photoDescription in adapter
     *
     * @param photos
     */
    public void setPhotoList(List<Uri> photos,List<String> photoDescription) {
            mPhotoList.clear();
            mPhotoDescription.clear();
            mPhotoList.addAll(photos);
            mPhotoDescription.addAll(photoDescription);
            notifyDataSetChanged();

    }
}

File: app/src/main/java/com/openclassrooms/realestatemanager/ui/createAndEditEstate/PhotoAdapter.kt
package com.openclassrooms.realestatemanager.ui.createAndEditEstate

import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.RequestManager
import com.openclassrooms.realestatemanager.databinding.ActivityAddPhotoItemBinding


class PhotoAdapter(
    listPhoto: MutableList<Uri>?,
    with: RequestManager,
    photoDescription: java.util.ArrayList<String>,
    estateEdit: Long
) : RecyclerView.Adapter<PhotoViewHolder>() {

    private val mPhotoList: MutableList<Uri> = ArrayList()
    private val mPhotoDescription: MutableList<String> = ArrayList()
    private val estateEdit: Long = 0

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PhotoViewHolder {
        val binding: ActivityAddPhotoItemBinding = ActivityAddPhotoItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return PhotoViewHolder(binding.root)
    }

    override fun onBindViewHolder(holder: PhotoViewHolder, position: Int) {
        //For photo description with photo
        var photoUri: Uri? = null
        var photoDescription = ""
        if (mPhotoList.size > position) {
            photoUri = mPhotoList[position]
        }
        if (mPhotoDescription.size > position) {
            photoDescription = mPhotoDescription[position]
        }
        try {
            holder.updateWithDetails(photoUri, photoDescription, estateEdit)
            notifyItemChanged(position);
        } catch (e: Exception) {
            e.message
        }
    }

    override fun getItemCount(): Int {
        return mPhotoList.size
    }

    /**
     * for set photolist in adapter
     *
     * @param photos
     */
    fun setPhotoList(photos: Collection<Uri>) {
        mPhotoList.clear()
        mPhotoList.addAll(photos)
        this.notifyDataSetChanged()
    }

    /**
     * For set photoDescription in adapter
     *
     * @param photoDescription
     */
    fun setPhotoDescription(photoDescription: Collection<String>) {
        mPhotoDescription.clear()
        mPhotoDescription.addAll(photoDescription)
        this.notifyDataSetChanged()
    }

}

class PhotoViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
    private val activityAddPhotoItemBinding: ActivityAddPhotoItemBinding? = null
    /**
     * For update with data
     *
     * @param photoList
     * @param photoDescription
     * @param estateEdit
     */
    fun updateWithDetails(photoList: Uri?, photoDescription: String?, estateEdit: Long) {
        activityAddPhotoItemBinding?.photoDescription?.setText(photoDescription)
        activityAddPhotoItemBinding?.photoImage?.setImageURI(photoList)

    }

}
